flipt-client-kotlin-android/src/main/java/io/flipt/client/FliptEvaluationClient.kt

```diff
@@ -0,0 +1,205 @@
+                namespace,
+                ClientOptions(
+                    url,
+                    updateInterval,
+                    authentication,
+                    reference,
+                    fetchMode
+                )
+            )
+        }
+    }
+
+
+    @Serializable
+    data class InternalEvaluationRequest(
+        @SerialName("flag_key") val flagKey: String,
+        @SerialName("entity_id") val entityId: String,
+        @SerialName("context") val context: Map<String, String>
+    )
+
+    fun evaluateVariant(
+        flagKey: String,
+        entityId: String,
+        context: Map<String, String>
+    ): VariantEvaluationResponse {
+        val evaluationRequest = InternalEvaluationRequest(flagKey, entityId, context)
+        val evaluationRequestSerialized = json.encodeToString(evaluationRequest)
+
+        val value = CLibrary.INSTANCE.evaluateVariant(engine, evaluationRequestSerialized)
+
+        val resp: Result<VariantEvaluationResponse> =
+            json.decodeFromString(Result.serializer(VariantEvaluationResponse.serializer()), value)
+
+        return resp.result ?: throw EvaluationException(resp.errorMessage ?: "Unknown Error")
+    }
+
+    fun evaluateBoolean(
+        flagKey: String,
+        entityId: String,
+        context: Map<String, String>
+    ): BooleanEvaluationResponse? {
+        val evaluationRequest = InternalEvaluationRequest(flagKey, entityId, context)
+        val evaluationRequestSerialized = json.encodeToString(evaluationRequest)
+
+        val value = CLibrary.INSTANCE.evaluateBoolean(engine, evaluationRequestSerialized)
+
+        val resp: Result<BooleanEvaluationResponse> =
+            json.decodeFromString(Result.serializer(BooleanEvaluationResponse.serializer()), value)
+
+        return resp.result ?: throw EvaluationException(resp.errorMessage ?: "Unknown Error")
+    }
+
+    fun evaluateBatch(batchEvaluationRequest: Array<EvaluationRequest>): BatchEvaluationResponse? {
+        val evaluationrequests = mutableListOf<InternalEvaluationRequest>()
+
+        batchEvaluationRequest.map {
+            evaluationrequests.add(InternalEvaluationRequest(it.flagKey, it.entityId, it.context))
+        }
+
+        val batchEvaluationRequestSerialized = json.encodeToString(evaluationrequests)
+        val value = CLibrary.INSTANCE.evaluateBatch(engine, batchEvaluationRequestSerialized)
+
+        val resp: Result<BatchEvaluationResponse> = json.decodeFromString(
+            Result.serializer(BatchEvaluationResponse.serializer()),
+            value
+        )
+
+        return resp.result ?: throw EvaluationException(resp.errorMessage ?: "Unknown Error")
+    }
+
+    fun listFlags(): ArrayList<Flag>? {
+        val value = CLibrary.INSTANCE.listFlags(engine)
+        val resp = readFlags(value)
+        return resp.result ?: throw EvaluationException(resp.errorMessage ?: "Unknown Error")
+    }
+
+    fun readFlags(ptr: String): Result<ArrayList<Flag>> {
```

Comment:
i dont think we want this function exposed publicly. can we just inline it in `listFlags`?